components/script/dom/bindings/trace.rs

```diff
@@ -160,6 +160,50 @@ impl<T: JSTraceable> JSTraceable for Rc<T> {
     }
 }
 
+impl<T: JSTraceable> JSTraceable for Arc<T> {
+    fn trace(&self, trc: *mut JSTracer) {
+        (**self).trace(trc)
+    }
+}
+
+impl JSTraceable for Mutex<Option<SharedRt>> {
```

Comment:
Well, I removed the ability of the macro to parse `<>` entirely, just to be on the safe side. Would you prefer I add it back or is this OK?